MapboxNavigation/ImageCache.swift

```diff
@@ -30,55 +19,22 @@ class ImageCache: BimodalImageCache {
         NotificationCenter.default.removeObserver(self)
     }
 
-    func store(_ image: UIImage, forKey key: String, toDisk: Bool, completion: NoArgBlock?) {
-        let key = cacheKeyForKey(key)
+    // MARK: Image cache
 
+    func store(_ image: UIImage, forKey key: String, toDisk: Bool, completion: CompletionHandler?) {
         storeImageInMemoryCache(image, forKey: key)
 
-        let dispatchCompletion = {
+        if toDisk == true {
+            if let data = UIImagePNGRepresentation(image) {
```

Comment:
@bsudekum yes. `UIImage` supports [several](https://developer.apple.com/library/content/documentation/2DDrawing/Conceptual/DrawingPrintingiOS/LoadingImages/LoadingImages.html) image data-types, and can represent itself as data in either a `PNG` or `JPG` format.

Reply:
@bsudekum yes. `UIImage` supports [several](https://developer.apple.com/library/content/documentation/2DDrawing/Conceptual/DrawingPrintingiOS/LoadingImages/LoadingImages.html) image data-types, and can represent itself as `Data` in either a `PNG` or `JPG` format.

Reply:
It's probably worth an integration test and/or manual verification though. This did feel like a bit of a shortcut.

Reply:
@JThramer I understand UIIMage supports many data formats, but I'm asking if this initializer supports more than just png, [which it looks like it does not](https://developer.apple.com/documentation/uikit/1624096-uiimagepngrepresentation). If the server one day changes to JPG, I think this will stop working. Should we try and investigate the data type before writing to disk?

Reply:
That function returns a PNG representation of the data. The data can also be represented in JPEG format. This is fine as-is but I'll add a test showing that the round-trip works.

Reply:
@bsudekum `UIImagePNGRepresentation(_:)` is a function to _serialize_ a `UIImage` object as a particular image representation in data. It supports just about any `UIImage` object you can throw at it, regardless of the format of the source data used to create the `UIImage`. You can verify this for yourself in a playground:

```swift
import UIKit
var imageObject = UIImage(named: "Aerial03.jpg") //any jpg image, or any supported image format
var pngData: Data? = UIImagePNGRepresentation(imageObject!) //does not fail
```

Reply:
Hopefully the test in [this commit](https://github.com/mapbox/mapbox-navigation-ios/pull/1226/commits/d5208c80fded42ab206236f36e0ab626d8007c6e) helps to clarify? If not, we can also consider adding a JPEG fixture to the project.